### PR TITLE
test: fix e2e tests broken in various ways

### PR DESCRIPTION
**Pull request by bermuell**
## Description

fix e2e tests broken due to various changes e.g. 
- cli command changes (query for reward in SDK module), 
- error message, checked in tests

Added imrovement for compatiblity tests to identify ICS version from git reference in a more generic way.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code 

## Summary by CodeRabbit

- **Chores**
	- Updated the `.gitignore` file to ignore files starting with `__debug_`, improving repository cleanliness.

- **Bug Fixes**
	- Enhanced command execution output by adding a `verbose` parameter to several command execution methods, offering more detailed logs.

- **Refactor**
	- Simplified command execution by removing the `ExecuteCommandWithVerbosity` function and integrating its logic directly into the existing `ExecuteCommand` function.
	- Improved error handling and version tag retrieval in the `getIcsVersion` function to enhance flexibility and robustness.

### CODE REVIEW

**Review by MSalopek — APPROVED** [review]
LGTM

File: tests/e2e/config.go [hunk 1]
```diff
@@ -128,15 +128,24 @@ func (tr *TestConfig) Initialize() {
...
 		return icsVersion
 	}
+
 	if semver.IsValid(reference) {
 		// remove build suffix
 		return semver.Canonical(reference)
 	}
 
-	for _, tag := range []string{"v2.0.0", "v2.4.0", "v2.4.0-lsm", "v3.1.0", "v3.2.0", "v3.3.0", "v4.0.0", "v4.1.1", "v4.1.1-lsm"} {
+	// List of all tags matching vX.Y.Z or vX.Y.X-lsm in ascending order
```
Comment (sainoe) [inline tests/e2e/config.go:141]:
```suggestion
	// List of all tags matching vX.Y.Z or vX.Y.Z-lsm in ascending order
```
(?)

**Comment by sainoe** [comment]
LGTM! 
QQ: does this fix the changeover e2e test?

**Comment by bermuell** [comment]
> LGTM! QQ: does this fix the changeover e2e test?

no, will use a different PR for that

**Pull request merged**